optimum/exporters/onnx/utils.py

```diff
@@ -158,15 +159,16 @@ def _get_submodels_for_export_stable_diffusion(
 
 
 def _get_submodels_for_export_decoder(
-    model: Union["PreTrainedModel", "TFPreTrainedModel"], use_past: bool
+    model: Union["PreTrainedModel", "TFPreTrainedModel"],
+    use_past: bool,
+    legacy: bool = False,
 ) -> Dict[str, Union["PreTrainedModel", "TFPreTrainedModel"]]:
     """
     Returns the decoder part of the model.
     """
-    models_for_export = {}
+    models_for_export = {ONNX_DECODER_NAME if legacy else "model": model}
 
-    models_for_export[ONNX_DECODER_NAME] = model
-    if use_past:
+    if legacy and use_past:
```

Comment:
How about encoder-decoders when legacy=False?

Reply:
this will impact only decoder models as this modification is done in `_get_submodels_for_export_decoder`, encoder-decoder models will not be impacted (not sure if that answers your question)

Reply:
Yes makes sense thank you!